lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -1228,164 +1237,201 @@ namespace Js
+        }
+        else if (sensitivity == CollatorSensitivity::Accent)
         {
-            return LINGUISTIC_IGNOREDIACRITIC | LINGUISTIC_IGNORECASE | NORM_IGNOREKANATYPE | NORM_IGNOREWIDTH;
+            flags |= LINGUISTIC_IGNORECASE | NORM_IGNOREKANATYPE | NORM_IGNOREWIDTH;
         }
-        else if (wcscmp(sensitivity, _u("accent")) == 0)
+        else if (sensitivity == CollatorSensitivity::Case)
         {
-            return LINGUISTIC_IGNORECASE | NORM_IGNOREKANATYPE | NORM_IGNOREWIDTH;
+            flags |= NORM_IGNOREKANATYPE | NORM_IGNOREWIDTH | LINGUISTIC_IGNOREDIACRITIC;
         }
-        else if (wcscmp(sensitivity, _u("case")) == 0)
+        else if (sensitivity == CollatorSensitivity::Variant)
         {
-            return  NORM_IGNOREKANATYPE | NORM_IGNOREWIDTH | LINGUISTIC_IGNOREDIACRITIC;
+            flags |= NORM_LINGUISTIC_CASING;
         }
-        else if (wcscmp(sensitivity, _u("variant")) == 0)
+
+        if (ignorePunctuation)
+        {
+            flags |= NORM_IGNORESYMBOLS;
+        }
+
+        if (numeric)
         {
-            return NORM_LINGUISTIC_CASING;
+            flags |= SORT_DIGITSASNUMBERS;
         }
-        return 0;
+
+        return flags;
     }
+#endif
 
-    // Takes arguments as follows(all required):
-    //     - [1] - String 1 for comparison
-    //     - [2] - String 2 for comparison
-    //     - [3] - Locale string (or undefined)
-    //     - [4] - Sensitivity string (or undefined)
-    //     - [5] - IgnorePunctuation boolean (or undefined)
-    //     - [6] - Numeric boolean (or undefined)
-    Var IntlEngineInterfaceExtensionObject::EntryIntl_CompareString(RecyclableObject* function, CallInfo callInfo, ...)
+    Var IntlEngineInterfaceExtensionObject::EntryIntl_CollatorGetCollation(RecyclableObject* function, CallInfo callInfo, ...)
     {
         EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
 
-        if (args.Info.Count < 7 || !JavascriptString::Is(args.Values[1]) || !JavascriptString::Is(args.Values[2]))
+#if defined(INTL_ICU)
+        if (args.Info.Count < 2 || !JavascriptString::Is(args.Values[1]))
         {
-            // CompareStringEx of undefined or non-strings is undefined
             return scriptContext->GetLibrary()->GetUndefined();
         }
 
-        DWORD compareFlags = 0;
-        JavascriptString* str1 = JavascriptString::FromVar(args.Values[1]);
-        JavascriptString* str2 = JavascriptString::FromVar(args.Values[2]);
+        JavascriptString *locale = JavascriptString::FromVar(args.Values[1]);
+        utf8::WideToNarrow locale8(locale->GetSz(), locale->GetLength());
 
-        WCHAR defaultLocale[LOCALE_NAME_MAX_LENGTH];
-        const char16 *givenLocale = nullptr;
-        defaultLocale[0] = '\0';
+        char collation8[LOCALE_NAME_MAX_LENGTH] = { 0 };
+        AssertMsg(CollatorGetCollation(locale8, collation8, _countof(collation8)), "Could not get resolved collation");
```

Comment:
>AssertMsg [](start = 8, length = 9)

What happens in release mode if this assert would have failed?

Reply:
Returning false basically means that we couldn't get the collation value from the string for an unexpected reason (rather than the expected reason of not having a collation value). So, its the same question about failfasts as before, though here we have a better idea of what the result in the failure case will be.